### ISSUE TITLE
Unhandled exception when sending assets from contract address

### ISSUE THREAD

**Issue by hal0x2328** [neo-project/neo#85]
Using the following test contract:
```
using Neo.SmartContract.Framework.Services.Neo;

namespace Neo.SmartContract
{
    public class Lock : Framework.SmartContract
    {
        public static bool Main()
        {
            return true;
        }
    }
}
```
If I understand correctly, this should allow anyone to spend from the contract address. But a transfer attempt fails on actual execution - the GUI always crashes with an exception "Index was outside the bounds of the array" in Neo.SmartContract.ContractParametersContext.Add.

**Comment by hal0x2328** [comment]
<img width="902" alt="screenshot 2017-10-20 09 43 13" src="https://user-images.githubusercontent.com/31932411/31824119-1cbd87ce-b57c-11e7-8024-c30f96733368.png">

**Comment by hal0x2328** [comment]
I suspected this could be related to main() not having any arguments since that also seems to be an issue with the verification code which automatically puts the signature data on the stack, so I tried again with another contract:
```
using Neo.SmartContract.Framework.Services.Neo;

namespace Neo.SmartContract
{
    public class Lock : Framework.SmartContract
    {
        public static bool Main(byte[] signature)
        {
            return true;
        }
    }
}
```
Same result was experienced however, exception "Index was outside the bounds of the array" in Neo.SmartContract.ContractParametersContext.Add

**Comment by birmas** [comment]
Hal, do you experience the same issue with your second example if you add the contract address with a parameter of 00? I've run through the logic here and determined that the index exception is being thrown when there is no parameter type of Signature. https://github.com/neo-project/neo/blob/3e1f3457cdc533bf6ea3e2887c39e2a9ddab768b/neo/SmartContract/ContractParametersContext.cs#L152

I've just successfully transferred funds out of a contract using this method. I believe neo.SmartContract.ContractParametersContext needs an update so that if index == -1 it doesn't try and proceed.

**Comment by birmas** [comment]
I think the code in ContractParametersContext needs to be reviewed. Verification fails if a SC has a Main definition such as Main(string op, params object[] args). When adding a contract address param type '00' tx will not be accepted by nodes

**Comment by birmas** [comment]
@hal0x2328 @erikzhang can this issue be closed with merge of #86 ?

**Comment by erikzhang** [comment]
It can be closed if someone has tested it.
